### PR TITLE
Fix GitHub hosted runner README links

### LINKED ISSUE #13336: Broken GitHub hosted runner README links
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

### What part(s) of the article would you like to see updated?

The README links listed under "For the overall list of included tools for each runner operating system, see the links below"

### Additional information

PR raised to fix this: https://github.com/github/docs/pull/13337

### PR DESCRIPTION
**Pull request by domdomegg**
### Why:

Closes #13336

### What's being changed:

Corrected broken links as they are case sensisitive.

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@domdomegg 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by muddylotus — COMMENTED** [review]
👍🏼

**Review by skedwards88 — APPROVED** [review]
Thanks for this fix! Our repos are currently frozen until early January, but we will merge this for you once the freeze lifts.

**Pull request merged**
